### ISSUE TITLE
Add plugin support for dbmigrate adapters & move Oracle to plugin

### ISSUE THREAD

**Issue by neokoenig** [cfwheels/cfwheels#711]
In order to get the dbmigrate branched merged into the core for 2.x, the intention is to drop Oracle support for dbmigrate (at least for now).

To make this easier, I'm planning on doing a simple check in `/plugins/` for any missing database adapter, and use that instead of attempting the load the local one if it exists.

This has the advantage that you could write an adapter plugin for an existing adapter easily.

At the moment, adapters are loaded via:
```
this.adapter = CreateObject("component","adapters.#dbType#");
```
in `wheels/dbmigrate/Migration.cfc`; each Adapter extends `wheels/dbmigrate/adapters/Abstract.cfc` ; so assuming the plugin CFC extended `wheels.dbmigrate.adapters.Abstract` would that work? Would it work in a subdir? Probably not. 

We could then perhaps add "dbmigrate" as a mixin to the plugins loadup script, so it wouldn't get exposed in odd places.

Then the adapter load process in Migration.cfc might look like:
```
// Pseudocode! Off the top of my head
this.adapterPluginPath = "/plugins/#dbType#/#dbType#.cfc";
this.adapterPath= fileExists(this.adapterPluginPath) ? this.adapterPluginPath : "adapters.#dbType#";
this.adapter = CreateObject("component", this.adapterPath ); 
```
The last thing we'd need to do is identify the db used at runtime and not run DBmigrate tests if it's oracle. Does that sound sensible or am I over/under thinking it?

**Comment by chapmandu** [comment]
Is there a `wheels` mapping in 2.0 that might work for apps in subdirectories?

**Comment by neokoenig** [comment]
Almost definitely. `wheelsmapping.` for one :)
This was written off the top of my head!

**Comment by neokoenig** [comment]
I'm tempted to leave this to post beta, we need user feedback first

**Comment by perdjurner** [comment]
Makes sense.

**Comment by perdjurner** [comment]
What's the situation / status here now that we no longer support Oracle?

**Comment by neokoenig** [comment]
Migrator still (ideally) needs plugin support at some point, but we should be able to simple remove the oracle adapter. I'd be happy to leave it to 2.1 TBH

**Comment by perdjurner** [comment]
Closing this as we're using the issue tracker for bugs only from now on.
Please implement your feature suggestion as a plugin or make a pull request.
If you need feedback before starting, please consult the Google Group.

Thanks.
